### ISSUE TITLE
Plugin prevents loading a page when the URL path ends with "svg"

### ISSUE THREAD

**Issue by ymeine** [jfgodoy/vite-plugin-solid-svg#40]
Hello,

Using this plugin prevents the Vite development server from loading the document/web page (the request with header `Sec-Fetch-Dest: document`) when the URL path ends with `svg`.

Examples:

- (re)load page on `/my-tab-to-demo-svg` fails
- (re)load page on `/my-tab-to-demo-canvas` works

## Error case

`ENOENT: no such file or directory, open 'C:\my-tab-to-demo-svg'`

Page displays: 

![image](https://github.com/jfgodoy/vite-plugin-solid-svg/assets/2077669/0435cde0-616d-488e-a11d-fe81e0078fe6)

## Working case

Page displays: 

![image](https://github.com/jfgodoy/vite-plugin-solid-svg/assets/2077669/76b12ddf-9555-44a2-869c-2dab530ea23b)

### Dependencies

```json
{
    "vite": "4.4.11",
    "vite-plugin-solid": "2.7.0",
    "vite-plugin-solid-svg": "0.6.4"
}
```

### Vite configuration file

```typescript
import {defineConfig} from 'vite';
import solid from 'vite-plugin-solid';
import solidSvg from 'vite-plugin-solid-svg';

export default defineConfig({
    plugins: [solid(), solidSvg()],
});
```

### Entry point for test

```typescript
document.body.innerText = location.pathname; // it loading succeeded, will show the URL path, simply
```

## Possible solutions

The first one would be to check for `.svg` instead of `svg`, unless I'm missing a rationale for not doing so.

One could argue that they would still want to have the document/web page load on a URL like `/my.tab.to.demo.svg` and that it would fail again then, but it actually fails much before on the Vite side, which apparently falls back to trying to load a physical entry point that would be located at `/my.tab.to.demo.svg` (fair enough, though it could check only for `html` extensions or simply fallback to default entry point, but I may miss more use cases with MPA and SSR).

Second solution would be to simply ignore any content that comes from a request with `Sec-Fetch-Dest: document` since it's supposed to load the application entry point, which I guess this plugin is not for. Btw, loading an SVG file from the `public` folder seems to not even go through the plugin at all anyways.

However, for that second solution, I'm not even sure there's a way to know the details of the request that triggered a pass through the plugins.

**Comment by jfgodoy** [comment]
Thank you, @ymeine, for the detailed report. I've resolved the issue, and it's now included in version 0.6.5. The solution that I chose was to check for `.svg` instead of `svg`.

**Comment by ymeine** [comment]
Thanks for the quick fix ! I think that's the pragmatic solution, especially since I'm not even sure the other one is feasible 😄
